indra/newview/pipeline.cpp

```diff
@@ -7133,6 +7147,50 @@ void LLPipeline::generateGlow(LLRenderTarget* src)
     }
 }
 
+void LLPipeline::applyCAS(LLRenderTarget* src, LLRenderTarget* dst)
+{
+    if (RenderCASSharpness == 0.0)
+    {
+        gPipeline.copyRenderTarget(src, dst);
+        return;
+    }
+
+    LLGLSLShader* sharpen_shader = &gCASProgram;
+
+    // Bind setup:
+    dst->bindTarget();
+
+    sharpen_shader->bind();
+
+    {
+        static LLStaticHashedString cas_param_0("cas_param_0");
+        static LLStaticHashedString cas_param_1("cas_param_1");
+        static LLStaticHashedString out_screen_res("out_screen_res");
+
```

Comment:
Not a blocker for a merge, but should probably just use a static LLCachedControl here for RenderCASSharpness.  "refreshCachedSettings" has a lot of side effects and is very slow, so unless a shader reload or a RenderTarget allocation is required, just use LLCachedControl wherever you want to access the saved setting.

Reply:
Fixed.